### DISCUSSION TITLE
Directory Picker

### DISCUSSION THREAD

**Discussion by GUOLDEV** [CommunityToolkit/Maui#172] [New Feature Discussions]
Implement Directory Picker, allowing the user to select a folder to save a file for example.

**Comment by TheCodeTraveler** [comment]
~We have a plan to add `FilePicker` from Xamarin.Essentials https://github.com/CommunityToolkit/Maui/issues/11~

Edit: `FilePicker` will be available in Maui.Essentials

**Comment by TheCodeTraveler** [comment]
Notes from [January Standup](https://www.youtube.com/watch?v=fp2tstk1_tw):
- Should this be implemented in Microsoft.Maui.Essentials?
- Would CommunityToolkit.Maui need to add  dependency to Microsoft.Maui.Essentials?
- Should we build this from scratch, not leveraging Microsoft.Maui.Essentials?

**Reply by VladislavAntonyuk** [reply]
1. I am voting for moving it to Maui repo and implementing it in Microsoft.Maui.Essentials
2. CommunityToolkit.Maui already uses Microsoft.Maui.Essentials (Snackbar gets current activity). It is included in the `<UseMaui>` tag implicitly.

**Comment by CliffAgius** [comment]
Closed as Answered
